examples/src/bin/entry_completion.rs

```diff
@@ -0,0 +1,109 @@
+//! # Entry completion example
+//!
+//! This example demonstrates how to build a list of items and use them
+//! to autocomplete a field as the user types in something.
+
+use gio::prelude::*;
+use gtk::prelude::*;
+
+use std::env::args;
+
+use gio::SimpleAction;
+use glib::Type;
+use gtk::{
+    Application, ApplicationWindow, Box, Entry, EntryCompletion, Label, ListStore, Orientation,
+};
+
+struct Data {
+    description: String,
+}
+
+fn create_list_model() -> ListStore {
+    let col_types: [Type; 1] = [Type::String];
+
+    let data: [Data; 4] = [
+        Data {
+            description: "France".to_string(),
+        },
+        Data {
+            description: "Italy".to_string(),
+        },
+        Data {
+            description: "Sweden".to_string(),
+        },
+        Data {
+            description: "Switzerland".to_string(),
+        },
+    ];
+    let store = ListStore::new(&col_types);
+    let col_indices: [u32; 1] = [0];
+    for d in data.iter() {
+        let values: [&dyn ToValue; 1] = [&d.description];
+        store.set(&store.append(), &col_indices, &values);
+    }
+    store
+}
+
+fn build_ui(application: &Application) {
+    // create the main window
+    let window = ApplicationWindow::new(application);
+    window.set_title("Entry with autocompletion");
+    window.set_default_size(840, 480);
+
+    // Create a title label
+    let win_title = Label::new(None);
+    win_title.set_markup("<big>Which country would you like to spend a holiday in?</big>");
+
+    // Create an EntryCompletion widget
+    let completion_countries = EntryCompletion::new();
+    // Use the first (and only) column available to set the autocompletion text
+    completion_countries.set_text_column(0);
+    // how many keystrokes to wait before attempting to autocomplete?
+    completion_countries.set_minimum_key_length(1);
+    // whether the completions should be presented in a popup window
+    completion_countries.set_popup_completion(true);
+
+    // Create a ListStore of items
+    // These will be the source for the autocompletion
+    // as the user types into the field
+    // For a more evolved example of ListStore see src/bin/list_store.rs
+    let ls = create_list_model();
+    completion_countries.set_model(Some(&ls));
```

Comment:
You could use a gtk::StringList here which is a Model itself. 
```suggestion
    let countries = gtk::StringList::new(&["France", "Switzerland", "Italy"]);
    completion_countries.set_model(Some(&countries));
```
the create_list_model & `Data` won't be needed. Other might changes might be needed as well

Reply:
looks like the above is not enough..

a StringList is a Model but EntryCompletion expects a TreeModel.
"pub fn set_model<P: IsA<TreeModel>>"

How do I generate a TreeModel from StringList?

Reply:
Agh I thought ComboBox used the new fancy stuff....